BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -706,12 +712,13 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didComp
         BOXLog(@"URLSessionTask %@ did finish loading. Requesting that the webview load the data (%lu bytes) with reponse %@", task, (unsigned long)[self.connectionData length], self.connectionResponse);
         [self setWebViewCanBeUsedDirectly:YES forHost:task.currentRequest.URL.host];
         [self setWebViewCanBeUsedDirectly:YES forHost:[self.connectionResponse URL].host];
-        [(UIWebView *)self.view loadData:self.connectionData
-                                MIMEType:[self.connectionResponse MIMEType]
-                        textEncodingName:[self.connectionResponse textEncodingName]
-                                 baseURL:[self.connectionResponse URL]];
-
-        self.connectionResponse = nil;
+        [BOXDispatchHelper callCompletionBlock:^{
+            [(UIWebView *)self.view loadData:self.connectionData
```

Comment:
same, this isn't a completion block call, it's just pushing some UI code to the main thread (always, regardless of whether things originally got called from main or not - which is what the dispatch helper is there to help with I believe)